File: curso_c_cpp/c/07_funciones/ejemplos/sumar.c
/*
 * Definir y usar funciones. Se muestran prototipos arriba y definiciones abajo.
 */
#include <stdio.h>

int sumar(int a, int b);           // prototipos
double areaCirculo(double radio);
void saludar(void);

int main(void) {
    saludar();
    printf("3 + 4 = %d\n", sumar(3, 4));
    printf("Area (r=2) = %.4f\n", areaCirculo(2.0));
    return 0;
}

int sumar(int a, int b) {
    return a + b;
}

double areaCirculo(double radio) {
    const double PI = 3.14159265;
    return PI * radio * radio;
}

void saludar(void) {
    printf("Bienvenido al ejemplo de funciones\n");
}

File: curso_java/07_metodos/ejemplos/Sumar.java
/*
 * Definir y usar metodos static. Incluye sobrecarga.
 */
public class Sumar {

    static int sumar(int a, int b) {
        return a + b;
    }
    static double areaCirculo(double radio) {
        final double PI = 3.14159265;
        return PI * radio * radio;
    }
    static void saludar(String nombre) {
        System.out.println("Hola, " + nombre);
    }

    public static void main(String[] args) {
        saludar("Java");
        System.out.println("3 + 4 = " + sumar(3, 4));
        System.out.printf("Area (r=2) = %.4f%n", areaCirculo(2.0));
    }
}
